tests/integration/__init__.py

```diff
@@ -166,6 +166,11 @@ def get_unused_localhost_port():
     usock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
     usock.bind(('127.0.0.1', 0))
     port = usock.getsockname()[1]
+    if port in (54505, 54506, 64505, 64506, 64510, 64511):
+        # These ports are hardcoded in the test configuration
+        usock.close()
+        return get_unused_localhost_port()
```

Comment:
There is nothing that prevents `get_unused_localhost_port()` from giving you a port in this range. In fact, if you just got a port in this range and you closed it, `get_unused_localhost_port()` is likely to give you back the same port.

Also, it is dangerous to hardcode ports that are in the ephemeral range since other processes totally unrelated to yours may be assigned the port via a similar mechanism that you are using. If you are hardcoding ports, I would suggest picking some ports < 32768 since those ports won't be assigned out to any asking process (just don't pick any ports of protocols that are in use on the system).